espflash/src/cli/mod.rs

```diff
@@ -277,7 +280,7 @@ pub struct MonitorConfigArgs {
     #[arg(long, requires = "non_interactive")]
     no_reset: bool,
     /// Logging format.
-    #[arg(long, short = 'L', default_value = "serial", requires = "elf")]
+    #[arg(long, short = 'L', default_value = "serial")]
```

Comment:
I removed the `requires` because it was failing when using it as a custom runner, since you have the elf from the image, but dont have the `elf` arg.

You still get a pretty decent error when you do `espflash monitor --log-format defmt` and dont provide and elf:
```
❯ cargo r -r --bin espflash -- monitor --log-format defmt
...
Commands:
    CTRL+R    Reset chip
    CTRL+C    Exit

Error: espflash::monitor::defmt::no_elf

  × No elf data available
  help: Please provide an ELF file with the `--elf` argument
```
But you get that after selecting the port

Reply:
Maybe we could investigate improving this in https://github.com/esp-rs/espflash/issues/764, since is somewhat similar

Reply:
It's also an issue in that `espflash flash <image> --elf <image>` is meaningless, but it's allowed.